Sara/tec 284 pre commit tasks (#2029)
semgrep/semgrep-docs: docs/extensions/pre-commit.md

---
slug: pre-commit
title: Run scans on pre-commit
hide_title: true
description: Learn to run a Semgrep scan before committing code; this prevents security issues or leaked secrets from entering your source control's history.
tags:
  - Extensions
---

import Login from "/src/components/procedure/_login-activate.mdx"

# Run scans on `pre-commit`

> The [pre-commit framework](https://pre-commit.com/) can run `semgrep` at commit-time. This is helpful to prevent secrets and security issues from leaking onto your Git history.

Reviewer: ```suggestion
The [pre-commit framework](https://pre-commit.com/) can run `semgrep` when you commit changes. This is helpful in preventing secrets and security issues from leaking into your Git history.
```
